### PR TITLE
Change: Service depot also reset breakdown chance.

### PR DESCRIPTION
**Pull request by Galigator**
I have so mush fun with this little change, that I really what to share it.
I have found the idea on Openttd forum.

Each time your vehicle go to service, then the breakdown chance is reset.
It allow to create large trains networks with simple tracks but carefully timed service and length to it.

Regards.

### CODE REVIEW

**Comment by nielsmh** [comment]
Hm so with this change, when you're playing with reduced breakdowns, vehicles will never break down right after receiving service?

**Comment by James103** [comment]
Suggested commit message (grammar fix): "Change: Service **at** depot reset**s** breakdown chance"

**Comment by ldpl** [comment]
This is just the same as the vehicle breaking down, doesn't protect it from further breakdowns completely, only makes it less likely for a while.

Though, I guess in some cases it makes it so much less likely that it's effectively 0.

**Comment by Galigator** [comment]
Yes you have too think it a bit like "predictive maintenance". If you do it well, everything goes well, if you start jamming your network sometime everything will go badly wrong after a while. Smoothness is require to operate an interconnection. Also train waiting in depot for an route, still start raise there breakdown chance, so your buffers must be carefully crafted; but you don't have to do silly/made network with logic to handle breakdown, or disable breakdown.
It just make it playable.


https://www.reddit.com/r/openttd/comments/4szlhj/an_easy_solution_for_fixing_breakdowns/

File: src/vehicle.cpp [hunk 1]
```diff
@@ -170,6 +170,8 @@ void VehicleServiceInDepot(Vehicle *v)
 		v->reliability = v->GetEngine()->reliability;
 		/* Prevent vehicles from breaking down directly after exiting the depot. */
 		v->breakdown_chance /= 4;
+		if (_settings_game.difficulty.vehicle_breakdowns == 1) // reduced breakdown.
+			v->breakdown_chance = 0;
```
Comment (michicc) [inline src/vehicle.cpp:174]: Coding style: Single line or {}

**Comment by Galigator** [comment]
#8317 correct and replace this PR.

**Comment by michicc** [comment]
Administrative note: A forced push to the old branch to update this PR would have been preferable, but this PR is small enough, so don't worry about it.

**Pull request closed without merging**